jans-linux-setup/jans_setup/setup_app/utils/base.py

```
@@ -381,20 +382,48 @@ def mylog(logs):
 
-    mylog("Downloading {} to {}".format(url, dst))
+    mylog(f"Downloading {url} to {dst}")
     download_tries = 1
+    download_ok = False
     while download_tries < 4:
         try:
-            urllib.request.urlretrieve(url, dst)
+            dst_tmp_fn = dst + '~' + os.urandom(4).hex()
+            urllib.request.urlretrieve(url, dst_tmp_fn)
+            shutil.move(dst_tmp_fn, dst)
             mylog("Download size: {} bytes".format(os.path.getsize(dst)))
```

Ruff (0.15.21)

[warning] Use f-string instead of `format` call

Convert to f-string

(UP032)